project/duties.py

```diff
@@ -263,6 +275,18 @@ def docs_deploy(ctx):
     """
     ctx.run("mkdocs gh-deploy", title="Deploying documentation")
 
+ 
+@duty
+def docs_deploy_gitlab(ctx):
+    """
+    Deploy the documentation on Gitlab Pages.
+
+    Arguments:
+        ctx: The context instance (passed automatically).
+    """
+    ctx.run("mkdocs build -s --site-dir public", title="Deploying documentation")
```

Comment:
Same comment as for the CI variables: it should be templated instead. It's not useful to have two actions to deploy to GitHub *and* GitLab pages. Users deploy to only one.

In that case, I would even argue that the site directory should be configured in `mkdocs.yml` itself instead of here!

Reply:
Put in PR for that here.
https://github.com/pawamoy/copier-poetry/pull/12

Awesome simple elegant solution :)

Reply:
I'd be happy to have this removed if we generalize docs_deploy from being Github specific
https://github.com/pawamoy/copier-poetry/blob/b5f0fb3210cbe6f91ac4d5fc65e3261613b96126/project/duties.py#L259

I think with https://github.com/pawamoy/copier-poetry/pull/12 and the latest tweaks in .gitlab-ci.yml it will work like you suggest and be least amount of bloat :)